### PR TITLE
Ban / Whitelist feature

### PR DESCRIPTION
**Pull request by greatman**
Player is kicked when he is banned or not in the whitelist.

### CODE REVIEW

File: src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java [hunk 1]
```diff
@@ -35,7 +39,39 @@
 public class BootstrapIdentificationMessageHandler extends MessageHandler<IdentificationMessage> {
 	@Override
 	public void handle(Session session, Player player, IdentificationMessage message) {
+		
+		//We check if the player is banned
+		String playerName = message.getName();
+		
+		Server s = (Server) session.getGame();	
+		
+		Iterator<Player> bannedPlayerList = s.getBannedPlayers().iterator();
```
Comment (Afforess) [inline src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java]: This is a terribly slow way to find out if a player is banned or whitelisted. What if there are 10000 players on the whitelist? The API needs to be changed to use Set, backed by a HashSet. Then you can do a contains call and have it be O(1) vs the current O(n) lookup time.

Comment (ZNickq) [inline src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java]: Agree...i'd do it, but that'll put multiple commits in my PR right now, and i suck at squashing them :P

Comment (stephenwilliams) [inline src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java]: Again @ZNickq if you do all of your PRs in separate branches you would be ok

Comment (ZNickq) [inline src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java]: Oh, that's an idea, netbeans lets me do that easily! :D
Ok, will do in the evening :)

Comment (windy1) [inline src/main/java/org/spout/vanilla/protocol/bootstrap/handler/BootstrapIdentificationMessageHandler.java]: git + netbeans = :(

**Comment by windy1** [comment]
Needs to be squashed and signed.

**Comment by windy1** [comment]
Also needs to be updated to current head which includes a new configuration.

**Comment by Salynne** [comment]
Closing these, open a new one if you've updated them.

**Comment by Zidane** [comment]
This will be closed so you can fix it and come back with a better pull

Thank you for your contribution regardless though...much appreciated :).

**Pull request closed without merging**